src/transports/HTTPTransport.ts

```diff
@@ -27,11 +29,36 @@ class HTTPTransport implements ITransport {
       this.onDataCallbacks.map((cb) => {
```

Comment:
I think the callback needs either an optional error field or to pass in sendError here, in the even that its a valid response 200, but malformed data.